docs/guides/unity-catalog.md

```diff
@@ -0,0 +1,530 @@
+    role_arn = aws_iam_role.external_data_access.arn
+  }
+  comment = "Managed by TF"
+}
+
+resource "databricks_grants" "external_creds" {
+  provider           = databricks.workspace
+  storage_credential = databricks_storage_credential.external.id
+  grant {
+    principal  = "Data Engineers"
+    privileges = ["CREATE TABLE"]
+  }
+}
+
+resource "databricks_external_location" "some" {
+  provider        = databricks.workspace
+  name            = "external"
+  url             = "s3://${aws_s3_bucket.external.id}/some"
+  credential_name = databricks_storage_credential.external.id
+  comment         = "Managed by TF"
+}
+
+resource "databricks_grants" "some" {
+  provider          = databricks.workspace
+  external_location = databricks_external_location.some.id
+  grant {
+    principal  = "Data Engineers"
+    privileges = ["CREATE TABLE", "READ FILES"]
+  }
+}
+```
+
+## Configure Unity Catalog clusters
+
+To ensure the integrity of ACLs, users are required to access Unity Catalog through compute resources configured with strong isolation guarantees and other security features. This is achieved through a new concept called ‘Security Modes’, which are applied to compute resources.
+
+To use Unity Catalog from a regular [databricks_cluster](../resources/cluster.md), one must configure the cluster with a proper **Data Security Mode**.  Currently, Unity Catalog is supported in two security modes: **User Isolation** and **Single User**.
+
+- **User Isolation** clusters can be shared by multiple users, but only SQL language is allowed. Some advanced cluster features such as library installation, init scripts and the DBFS Fuse mount are also disabled in this mode to ensure security isolation among cluster users.
+
+```hcl
+data "databricks_spark_version" "latest" {
+  provider = databricks.workspace
+}
+data "databricks_node_type" "smallest" {
+  provider   = databricks.workspace
+  local_disk = true
+}
+
+resource "databricks_cluster" "unity_sql" {
+  provider                = databricks.workspace
+  cluster_name            = "Unity SQL"
+  spark_version           = data.databricks_spark_version.latest.id
+  node_type_id            = data.databricks_node_type.smallest.id
+  autotermination_minutes = 60
+  enable_elastic_disk     = false
+  num_workers             = 2
+  aws_attributes {
+    availability = "SPOT"
+  }
+  data_security_mode = "USER_ISOLATION"
+}
+```
+
+- To use those advanced cluster features or languages like Python, Scala and R with Unity Catalog, one must choose **Single User** Mode when launching the cluster. The cluster can only be used exclusively by a single user (by default the owner of the cluster); other users are not allowed to attach to the cluster.
+This means a group of users, which is managed as a group through SCIM provisioning, will a collection of single-user [databricks_cluster](../resources/cluster.md), which they should be able to restart. Terraform's `for_each` meta-attribute helps to do this easily.
```

Comment:
I don't understand this sentence.

Reply:
@nkvuong Meant to tell that "let terraform create a cluster for every user in the group"